.github/workflows/test-upstream.yml

```diff
@@ -173,59 +193,32 @@ jobs:
     steps:
       - uses: actions/checkout@v2
-      - name: Report failures
-        uses: actions/github-script@v3
+      - uses: actions/download-artifact@v3
+      - name: Prepare logs & issue label
+        run: |
+          # import testing doesn't generate a pytest log but one is needed for xarray-contrib/issue-from-pytest-log
+          if ls test*.jsonl 1> /dev/null 2>&1; then
+            paste -d"\n" test*.jsonl > results.jsonl
+          else
+            touch results.jsonl
+          fi
+
+          # convert which_upstream to lowercase
+          echo "which_upstream_lower=${which_upstream,,}" >> $GITHUB_ENV
+      - name: Open or update issue on failure
+        uses: xarray-contrib/issue-from-pytest-log@v1.2.6
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          script: |
-            const title = "⚠️ Upstream CI ${{ env.which_upstream }} failed ⚠️"
-            const workflow_url = `https://github.com/${process.env.GITHUB_REPOSITORY}/actions/runs/${process.env.GITHUB_RUN_ID}`
-            const issue_body = `[Workflow Run URL](${workflow_url})`
-            // Run GraphQL query against GitHub API to find the most recent open issue used for reporting failures
-            const query = `query($owner:String!, $name:String!, $creator:String!, $label:String!){
-              repository(owner: $owner, name: $name) {
-                issues(first: 1, states: OPEN, filterBy: {createdBy: $creator, labels: [$label]}, orderBy: {field: CREATED_AT, direction: DESC}) {
-                  edges {
-                    node {
-                      body
-                      id
-                      number
-                    }
-                  }
-                }
-              }
-            }`;
-            const variables = {
-                owner: context.repo.owner,
-                name: context.repo.repo,
-                label: 'upstream',
-                creator: "github-actions[bot]"
-            }
-            const result = await github.graphql(query, variables)
-            // If no issue is open, create a new issue,
-            // else update the body of the existing issue.
-            if (result.repository.issues.edges.length === 0) {
-                github.issues.create({
-                    owner: variables.owner,
-                    repo: variables.name,
-                    body: issue_body,
-                    title: title,
-                    labels: [variables.label]
-                })
-            } else {
-                github.issues.update({
-                    owner: variables.owner,
-                    repo: variables.name,
-                    issue_number: result.repository.issues.edges[0].node.number,
-                    body: issue_body
-                })
-            }
+          log-path: results.jsonl
+          issue-title: ⚠️ Upstream CI ${{ env.which_upstream }} failed ⚠️
+          issue-label: upstream, ${{ env.which_upstream_lower }}
```

Comment:
From the action docs it seems like we should be able to specify multiple labels here, but haven't seen an example of this so not sure if this will break things. An alternative could be to just split off our `upstream` label into `upstream-dask` and `upstream-datafusion`

Reply:
I think splitting off the labels is a good solution for now